pkg/registries/registries_test.go

```diff
@@ -497,4 +1021,117 @@ func TestEditRegistriesConfig(t *testing.T) {
 			assert.NoError(t, err)
 		})
 	}
+
+	// should fail if idms or itms exist with icsp
+	failtests := []struct {
+		name              string
+		insecure, blocked []string
+		idmsRules         []*apicfgv1.ImageDigestMirrorSet
+		itmsRules         []*apicfgv1.ImageTagMirrorSet
+		icspRules         []*apioperatorsv1alpha1.ImageContentSourcePolicy
+		want              sysregistriesv2.V2RegistriesConf
```

Comment:
`insecure`, `blocked`, `want` are unused, so that can be dropped.

Non-blocking: This test could be shorter and somewhat easier to fit on the screen with:
- Only including one set of mirrors instead of three in every item.
- Making the test data `idmsNonempty, itmsNonempty, icspNonempty bool`, and letting the testing loop provide data or an empty array.